app/services/integration/kubernetes_service.rb

```diff
@@ -178,7 +237,30 @@ def build_routes(name, specs = [], owner: get_owner, **metadata)
     end
   end
 
+  def build_ingresses(name, specs = [], owner: get_owner, **metadata)
+    specs.map do |spec|
+      Ingress.new(
+        metadata: {
+          generateName: name,
+          namespace: namespace,
+          labels: owner.metadata.labels,
+          ownerReferences: [as_reference(owner)]
+        }.deep_merge(metadata.deep_merge(
+          labels: {
+            'zync.3scale.net/route-to': spec.to_h.dig(:rules, 0, :http, :paths, 0, :backend, :service, :name),
+          },
+          annotations: {
+            'zync.3scale.net/host': spec.host,
+            'kubernetes.io/ingress.class': Rails.application.config.integrations.fetch(:kubernetes_ingress_class, 'nginx'),
```

Comment:
Since the PR was open, this annotation has been deprecated :upside_down_face: 
https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation

Reply:
Yea you are right, I will move it to the correct location in spec.ingressClassName